### PR TITLE
Abstract side content

### PR DESCRIPTION
**Pull request by ningyuansg**
Blocked by #89.

---

### Features

- The `SideContent` component is now a 'higher order' component
- SideContent renders itself according to an array in the props passed to it
- The array has elements with properties label, icon, and body
- The body is a JSX.Element to be rendered as side content
- If the array length is greater than 1, a header is rendered with buttons (button icon from property icon, tooltip with content from property label) to switch between bodies
- The header is not rendered if the array is size 1
- The active 'tab' being shown in the side content is persistent between workspace mount/dismount
- These 'content props' to the `SideContent` component should be passed through the Workspace component rather than directly

### CODE REVIEW

File: src/styles/_workspace.scss [hunk 1]
```diff
@@ -106,9 +108,22 @@ $code-color-error: #ff4444;
 
     .Editor {
       flex: 1 1 auto;
+      margin: 0 0.5rem 0 0.5rem;
+      padding: 0;
     }
   }
 
+  .side-content-header {
```
Comment (remo5000) [inline src/styles/_workspace.scss]: This needs to wrap when overflowing (for mobile users / moving the resizable editor). I'll fix it, should be one property.

Comment (remo5000) [reply src/styles/_workspace.scss]: Done

File: src/components/Playground.tsx [hunk 2]
```diff
@@ -15,4 +17,32 @@ const keyMap = {
   goGreen: 'h u l k'
 }
 
+const SICP_SITE = 'http://www.comp.nus.edu.sg/~henz/sicp_js/'
+const CHAP = '\xa7'
+const playgroundIntroduction = {
```
[the remark this answers is not in the document]
Comment (remo5000) [reply src/components/Playground.tsx]: Done!

File: src/reducers/playground.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { Reducer } from 'redux'
 import {
+  CHANGE_ACTIVE_TAB,
```
[the remark this answers is not in the document]
Comment (ningyuansg) [reply src/reducers/playground.ts]: There are no other tabs in the workspace, so even if we do not specify that the tab belongs to the side content, there should be little ambiguity.

[the remark this answers is not in the document]
Comment (remo5000) [reply src/reducers/playground.ts]: Sure, we'll change it in the future if there are some other tabs

**Pull request merged**